File: Node.cpp
/**
 * Node.cpp - Assignment01
 * Author: Cody Lewis
 * Student No. 3283349
 * Date: 2017-08-16
 * Last modified: 2017-08-24
 * Description:
 * Defines the Nodes of the LinkedList
 */
#include<iostream>
#include "Node.h"
using namespace std;
namespace lewis_a01_Node{
  Node::Node(){
    prvPtr = NULL;  //Initially the pointers must be set
    nxtPtr = NULL;  //as NULL
  };
  Node::Node(const item& itm){
    content = itm;
    prvPtr = NULL;
    nxtPtr = NULL;
  };
  void Node::setContent(const item& cntnt){
    content = cntnt;
  };
  void Node::setPrev(Node* prev){
    prvPtr = prev;
  };
  void Node::setNext(Node* next){
    nxtPtr = next;
  };
  Node::item Node::getContent() const{
    return content;
  };
  Node* Node::getPrev() const{
    return prvPtr;
  };
  Node* Node::getNext() const{
    return nxtPtr;
  };
}

File: DataStructures/Map/Node.java
/* Author: Cody Lewis
 * Date:   18-APR-2018
 */
public class Node<T>{
  // Members
  private T data;
  private Node<T> prev;
  private Node<T> next;
  // Constructors
  public Node(){
    data = null;
    prev = null;
    next = null;
  }
  public Node(T data,Node<T> prev,Node<T> next){
    this.data = data;
    this.prev = prev;
    this.next = next;
  }
  // Mutators
  public void setData(T data){ this.data = data; }
  public void setPrev(Node<T> prev){ this.prev = prev; }
  public void setNext(Node<T> next){ this.next = next; }
  // Queries
  public T getData(){ return data; }
  public Node<T> getPrev(){ return prev; }
  public Node<T> getNext(){ return next; }
}
